spring-boot/admin/src/main/java/eu/coatrack/admin/controllers/AdminController.java

```diff
@@ -333,6 +334,34 @@ public DoughnutChart generateUserStatisticsDoughnutChart(
         }
     }
 
+
+	@RequestMapping(value = "/dashboard/serviceStatsChartCanvas", method = GET, produces = "application/json")
+    @ResponseBody
+    public DoughnutChart generateServiceStatisticsDoughnutChart(
+            @RequestParam("dateFrom") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate selectedTimePeriodStart,
+            @RequestParam("dateUntil") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate selectedTimePeriodEnd) {
+
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        List<StatisticsPerService> userStatsList = metricsAggregationCustomRepository
```

Comment:
Only one statement at once. Don't interlace several statements. This would be easier to read:
```java
List<ApiService>servicesOwnedByLoggedInUser = serviceApiRepository.findByOwnerUsername(auth.getName());
List<StatisticsPerService> userStatsList = metricsAggregationCustomRepository
                .getStatisticsOfServicesOwnedByLoggedInUserAndByDateRange(
                        servicesOwnedByLoggedInUser, selectedTimePeriodStart, selectedTimePeriodEnd);
```

Reply:
What do you mean with interlace several statements?

Reply:
What do you mean with interlace several statements?

Reply:
Done and lesson learned ;)